### PR TITLE
Tweak Vite config

### PR DESCRIPTION
**Pull request by nstepien**

### CODE REVIEW

File: vite.config.ts [hunk 1]
```diff
@@ -19,38 +20,39 @@ const resizeColumn: BrowserCommand<[resizeBy: number]> = async (context, resizeB
...
-  base: isCI ? '/react-data-grid/' : '/',
+export default defineConfig(({ command }) => ({
+  base: '/react-data-grid/',
   cacheDir: '.cache/vite',
+  clearScreen: false,
   build: {
-    emptyOutDir: true,
-    sourcemap: true
+    target: browserslist().map((version) => version.replace(' ', '')),
+    modulePreload: { polyfill: false },
```
Comment (nstepien) [inline vite.config.ts:29]: This will optimize the build output

File: vite.config.ts [hunk 2]
```diff
@@ -19,38 +20,39 @@ const resizeColumn: BrowserCommand<[resizeBy: number]> = async (context, resizeB
   await page.mouse.up();
 };
 
-export default defineConfig({
-  base: isCI ? '/react-data-grid/' : '/',
+export default defineConfig(({ command }) => ({
+  base: '/react-data-grid/',
```
Comment (nstepien) [inline vite.config.ts:24]: Keeping it consistent between dev/prod

File: vite.config.ts [hunk 3]
```diff
@@ -19,38 +20,39 @@ const resizeColumn: BrowserCommand<[resizeBy: number]> = async (context, resizeB
...
   },
   test: {
-    root: '.',
     globals: true,
     coverage: {
       provider: 'istanbul',
       enabled: isCI,
-      include: ['src/**/*.{ts,tsx}', '!src/types.ts'],
-      reporter: ['text', 'json']
+      include: ['src/**/*.{ts,tsx}'],
```
Comment (nstepien) [inline vite.config.ts:54]: Vitest should now handle type-only files well.

File: vite.config.ts (hunk 1 above)
Comment (amanmahajan7) [reply vite.config.ts:29]: Which polyfills were getting included in the build?

**Review by amanmahajan7 — APPROVED** [review]

File: vite.config.ts (hunk 1 above)
Comment (nstepien) [reply vite.config.ts:29]: Nothing in the RDG output, but I remember seeing differences in other projects.
I think it'd matter if we had lazy-loaded modules, so it'd use native methods to pre-load split chunks.

**Pull request merged**